dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/CompositionHost.cs

```diff
@@ -107,29 +95,31 @@ private object InitializeCoreDispatcher()
         }
 
         // Get compositor and target for hwnd.
-        private void InitComposition(IntPtr hwndHost)
+        private void InitializeComposition(IntPtr hwndHost)
         {
-            ICompositorDesktopInterop interop;
-
-            this.Compositor = new Compositor();
-            object iunknown = Compositor as object;
-            interop = (ICompositorDesktopInterop)iunknown;
-            interop.CreateDesktopWindowTarget(hwndHost, true, out IntPtr raw);
-
-            object rawObject = Marshal.GetObjectForIUnknown(raw);
-            compositionTarget = (ICompositionTarget)rawObject;
-
-            if (raw == null) { throw new Exception("QI Failed"); }
-
+            Compositor = new Compositor();
+            var compositorDesktopInterop = (ICompositorDesktopInterop)(object)Compositor;
+            compositorDesktopInterop.CreateDesktopWindowTarget(hwndHost, true, out _compositionTarget);
         }
 
         protected override void DestroyWindowCore(HandleRef hwnd)
```

Comment:
>DestroyWindowCore [](start = 32, length = 17)

Are you sure this only ever gets called once? Maybe the window can be created and destroyed multiple times on the one HwndHost.

Reply:
Debugging and going through various scenarios this appears to be the case unless you know otherwise?

---
In reply to: [276434344](https://github.com/Microsoft/Windows.UI.Composition-Win32-Samples/pull/1#discussion_r276434344) [](ancestors = 276434344)